File: api/src/main/java/kr/toxicity/model/api/manager/GlobalManager.java
package kr.toxicity.model.api.manager;

import kr.toxicity.model.api.pack.PackZipper;
import org.jetbrains.annotations.NotNull;

/**
 * Global manager
 */
public interface GlobalManager {
    /**
     * Executes reload
     * @param info reload info
     * @param zipper pack zipper
     */
    void reload(@NotNull ReloadInfo info, @NotNull PackZipper zipper);
}

File: core/src/main/kotlin/kr/toxicity/model/manager/GlobalManager.kt
/**
 * This source file is part of BetterModel.
 * Copyright (c) 2024–2025 toxicity188
 * Licensed under the MIT License.
 * See LICENSE.md file for full license text.
 */
package kr.toxicity.model.manager

import kr.toxicity.model.api.pack.PackZipper

interface GlobalManager {
    fun start() {}
    fun reload(pipeline: ReloadPipeline, zipper: PackZipper)
    fun end() {}
}